src/odemis/acq/align/autofocus.py

```diff
@@ -660,13 +660,12 @@ def Sparc2AutoFocus(align_mode, opm, streams=None, start_autofocus=True):
         raise LookupError("Failed to focus: %s", ex)
 
     detector = None
-    if isinstance(streams, (list,)):
-        for s in streams:
-            for d in dets:
-                if d.role == s.detector.role:
-                    detector = s.detector
-            if detector == None:
-                logging.warning("The detector of the stream is not found to be one of the picked detectors %s")
+    for s in streams:
+        for d in dets:
```

Comment:
As this is just to show a warning when the detector is not found, you could actually just write:
```
for s in stream:
    if s.detector.role not in (d.role for d in dets):
        logging.....
```